From 6509189d8a6073abde756e59479759bf5e12f608 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Sun, 28 Apr 2019 16:30:19 -0400
Subject: [PATCH 1/2] Qt: remove autogenerated files from the tree

These are always wrong and should be regenerated with qmake/moc.
---
 src/GUI/Makefile    | 220 --------------------------------------------
 src/GUI/i7z_GUI.moc |  93 -------------------
 2 files changed, 313 deletions(-)
 delete mode 100644 src/GUI/Makefile
 delete mode 100644 src/GUI/i7z_GUI.moc

diff --git a/src/GUI/Makefile b/src/GUI/Makefile
deleted file mode 100644
index 3bfbfda..0000000
--- a/src/GUI/Makefile
+++ /dev/null
@@ -1,220 +0,0 @@
-#############################################################################
-# Makefile for building: i7z_GUI
-# Generated by qmake (2.01a) (Qt 4.8.4) on: Fri Sep 13 10:41:57 2013
-# Project:  i7z_GUI.pro
-# Template: app
-# Command: /usr/lib/x86_64-linux-gnu/qt4/bin/qmake -o Makefile i7z_GUI.pro
-#############################################################################
-
-####### Compiler, tools and options
-
-CC            = gcc
-CXX           = g++
-DEFINES       = -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED
-CFLAGS        = -m64 -pipe -v -fno-schedule-insns2 -fno-schedule-insns -fno-inline-small-functions -fno-caller-saves -g -Wall -W -D_REENTRANT $(DEFINES)
-CXXFLAGS      = -m64 -pipe -v -fno-schedule-insns2 -fno-schedule-insns -fno-inline-small-functions -fno-caller-saves -g -Wall -W -D_REENTRANT $(DEFINES)
-INCPATH       = -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4 -I. -I.
-LINK          = g++
-LFLAGS        = -m64
-LIBS          = $(SUBLIBS)  -L/usr/lib/x86_64-linux-gnu -lQtGui -lQtCore -lpthread 
-AR            = ar cqs
-RANLIB        = 
-QMAKE         = /usr/lib/x86_64-linux-gnu/qt4/bin/qmake
-TAR           = tar -cf
-COMPRESS      = gzip -9f
-COPY          = cp -f
-SED           = sed
-COPY_FILE     = $(COPY)
-COPY_DIR      = $(COPY) -r
-STRIP         = strip
-INSTALL_FILE  = install -m 644 -p
-INSTALL_DIR   = $(COPY_DIR)
-INSTALL_PROGRAM = install -m 755 -p
-DEL_FILE      = rm -f
-SYMLINK       = ln -f -s
-DEL_DIR       = rmdir
-MOVE          = mv -f
-CHK_DIR_EXISTS= test -d
-MKDIR         = mkdir -p
-
-####### Output directory
-
-OBJECTS_DIR   = ./
-
-####### Files
-
-SOURCES       = i7z_GUI.cpp 
-OBJECTS       = i7z_GUI.o
-DIST          = /usr/share/qt4/mkspecs/common/unix.conf \
-		/usr/share/qt4/mkspecs/common/linux.conf \
-		/usr/share/qt4/mkspecs/common/gcc-base.conf \
-		/usr/share/qt4/mkspecs/common/gcc-base-unix.conf \
-		/usr/share/qt4/mkspecs/common/g++-base.conf \
-		/usr/share/qt4/mkspecs/common/g++-unix.conf \
-		/usr/share/qt4/mkspecs/qconfig.pri \
-		/usr/share/qt4/mkspecs/features/qt_functions.prf \
-		/usr/share/qt4/mkspecs/features/qt_config.prf \
-		/usr/share/qt4/mkspecs/features/exclusive_builds.prf \
-		/usr/share/qt4/mkspecs/features/default_pre.prf \
-		/usr/share/qt4/mkspecs/features/debug.prf \
-		/usr/share/qt4/mkspecs/features/default_post.prf \
-		/usr/share/qt4/mkspecs/features/unix/gdb_dwarf_index.prf \
-		/usr/share/qt4/mkspecs/features/warn_on.prf \
-		/usr/share/qt4/mkspecs/features/qt.prf \
-		/usr/share/qt4/mkspecs/features/unix/thread.prf \
-		/usr/share/qt4/mkspecs/features/moc.prf \
-		/usr/share/qt4/mkspecs/features/resources.prf \
-		/usr/share/qt4/mkspecs/features/uic.prf \
-		/usr/share/qt4/mkspecs/features/yacc.prf \
-		/usr/share/qt4/mkspecs/features/lex.prf \
-		/usr/share/qt4/mkspecs/features/include_source_dir.prf \
-		i7z_GUI.pro
-QMAKE_TARGET  = i7z_GUI
-DESTDIR       = 
-TARGET        = i7z_GUI
-
-first: all
-####### Implicit rules
-
-.SUFFIXES: .o .c .cpp .cc .cxx .C
-
-.cpp.o:
-	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
-
-.cc.o:
-	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
-
-.cxx.o:
-	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
-
-.C.o:
-	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o "$@" "$<"
-
-.c.o:
-	$(CC) -c $(CFLAGS) $(INCPATH) -o "$@" "$<"
-
-####### Build rules
-
-all: Makefile $(TARGET)
-
-$(TARGET):  $(OBJECTS)  
-	$(LINK) $(LFLAGS) -o $(TARGET) $(OBJECTS) $(OBJCOMP) $(LIBS)
-	{ test -n "$(DESTDIR)" && DESTDIR="$(DESTDIR)" || DESTDIR=.; } && test $$(gdb --version | sed -e 's,[^0-9]\+\([0-9]\)\.\([0-9]\).*,\1\2,;q') -gt 72 && gdb --nx --batch --quiet -ex 'set confirm off' -ex "save gdb-index $$DESTDIR" -ex quit '$(TARGET)' && test -f $(TARGET).gdb-index && objcopy --add-section '.gdb_index=$(TARGET).gdb-index' --set-section-flags '.gdb_index=readonly' '$(TARGET)' '$(TARGET)' && rm -f $(TARGET).gdb-index || true
-
-Makefile: i7z_GUI.pro  /usr/share/qt4/mkspecs/linux-g++-64/qmake.conf /usr/share/qt4/mkspecs/common/unix.conf \
-		/usr/share/qt4/mkspecs/common/linux.conf \
-		/usr/share/qt4/mkspecs/common/gcc-base.conf \
-		/usr/share/qt4/mkspecs/common/gcc-base-unix.conf \
-		/usr/share/qt4/mkspecs/common/g++-base.conf \
-		/usr/share/qt4/mkspecs/common/g++-unix.conf \
-		/usr/share/qt4/mkspecs/qconfig.pri \
-		/usr/share/qt4/mkspecs/features/qt_functions.prf \
-		/usr/share/qt4/mkspecs/features/qt_config.prf \
-		/usr/share/qt4/mkspecs/features/exclusive_builds.prf \
-		/usr/share/qt4/mkspecs/features/default_pre.prf \
-		/usr/share/qt4/mkspecs/features/debug.prf \
-		/usr/share/qt4/mkspecs/features/default_post.prf \
-		/usr/share/qt4/mkspecs/features/unix/gdb_dwarf_index.prf \
-		/usr/share/qt4/mkspecs/features/warn_on.prf \
-		/usr/share/qt4/mkspecs/features/qt.prf \
-		/usr/share/qt4/mkspecs/features/unix/thread.prf \
-		/usr/share/qt4/mkspecs/features/moc.prf \
-		/usr/share/qt4/mkspecs/features/resources.prf \
-		/usr/share/qt4/mkspecs/features/uic.prf \
-		/usr/share/qt4/mkspecs/features/yacc.prf \
-		/usr/share/qt4/mkspecs/features/lex.prf \
-		/usr/share/qt4/mkspecs/features/include_source_dir.prf \
-		/usr/lib/x86_64-linux-gnu/libQtGui.prl \
-		/usr/lib/x86_64-linux-gnu/libQtCore.prl
-	$(QMAKE) -o Makefile i7z_GUI.pro
-/usr/share/qt4/mkspecs/common/unix.conf:
-/usr/share/qt4/mkspecs/common/linux.conf:
-/usr/share/qt4/mkspecs/common/gcc-base.conf:
-/usr/share/qt4/mkspecs/common/gcc-base-unix.conf:
-/usr/share/qt4/mkspecs/common/g++-base.conf:
-/usr/share/qt4/mkspecs/common/g++-unix.conf:
-/usr/share/qt4/mkspecs/qconfig.pri:
-/usr/share/qt4/mkspecs/features/qt_functions.prf:
-/usr/share/qt4/mkspecs/features/qt_config.prf:
-/usr/share/qt4/mkspecs/features/exclusive_builds.prf:
-/usr/share/qt4/mkspecs/features/default_pre.prf:
-/usr/share/qt4/mkspecs/features/debug.prf:
-/usr/share/qt4/mkspecs/features/default_post.prf:
-/usr/share/qt4/mkspecs/features/unix/gdb_dwarf_index.prf:
-/usr/share/qt4/mkspecs/features/warn_on.prf:
-/usr/share/qt4/mkspecs/features/qt.prf:
-/usr/share/qt4/mkspecs/features/unix/thread.prf:
-/usr/share/qt4/mkspecs/features/moc.prf:
-/usr/share/qt4/mkspecs/features/resources.prf:
-/usr/share/qt4/mkspecs/features/uic.prf:
-/usr/share/qt4/mkspecs/features/yacc.prf:
-/usr/share/qt4/mkspecs/features/lex.prf:
-/usr/share/qt4/mkspecs/features/include_source_dir.prf:
-/usr/lib/x86_64-linux-gnu/libQtGui.prl:
-/usr/lib/x86_64-linux-gnu/libQtCore.prl:
-qmake:  FORCE
-	@$(QMAKE) -o Makefile i7z_GUI.pro
-
-dist: 
-	@$(CHK_DIR_EXISTS) .tmp/i7z_GUI1.0.0 || $(MKDIR) .tmp/i7z_GUI1.0.0 
-	$(COPY_FILE) --parents $(SOURCES) $(DIST) .tmp/i7z_GUI1.0.0/ && $(COPY_FILE) --parents i7z_GUI.cpp .tmp/i7z_GUI1.0.0/ && (cd `dirname .tmp/i7z_GUI1.0.0` && $(TAR) i7z_GUI1.0.0.tar i7z_GUI1.0.0 && $(COMPRESS) i7z_GUI1.0.0.tar) && $(MOVE) `dirname .tmp/i7z_GUI1.0.0`/i7z_GUI1.0.0.tar.gz . && $(DEL_FILE) -r .tmp/i7z_GUI1.0.0
-
-
-clean:compiler_clean 
-	-$(DEL_FILE) $(OBJECTS)
-	-$(DEL_FILE) *~ core *.core
-
-
-####### Sub-libraries
-
-distclean: clean
-	-$(DEL_FILE) $(TARGET) 
-	-$(DEL_FILE) Makefile
-
-
-check: first
-
-mocclean: compiler_moc_header_clean compiler_moc_source_clean
-
-mocables: compiler_moc_header_make_all compiler_moc_source_make_all
-
-compiler_moc_header_make_all:
-compiler_moc_header_clean:
-compiler_rcc_make_all:
-compiler_rcc_clean:
-compiler_image_collection_make_all: qmake_image_collection.cpp
-compiler_image_collection_clean:
-	-$(DEL_FILE) qmake_image_collection.cpp
-compiler_moc_source_make_all: i7z_GUI.moc
-compiler_moc_source_clean:
-	-$(DEL_FILE) i7z_GUI.moc
-i7z_GUI.moc: ../helper_functions.c \
-		../i7z.h \
-		i7z_GUI.cpp
-	/usr/lib/x86_64-linux-gnu/qt4/bin/moc $(DEFINES) $(INCPATH) i7z_GUI.cpp -o i7z_GUI.moc
-
-compiler_uic_make_all:
-compiler_uic_clean:
-compiler_yacc_decl_make_all:
-compiler_yacc_decl_clean:
-compiler_yacc_impl_make_all:
-compiler_yacc_impl_clean:
-compiler_lex_make_all:
-compiler_lex_clean:
-compiler_clean: compiler_moc_source_clean 
-
-####### Compile
-
-i7z_GUI.o: i7z_GUI.cpp ../helper_functions.c \
-		../i7z.h \
-		i7z_GUI.moc
-	$(CXX) -c $(CXXFLAGS) $(INCPATH) -o i7z_GUI.o i7z_GUI.cpp
-
-####### Install
-
-install:   FORCE
-
-uninstall:   FORCE
-
-FORCE:
-
diff --git a/src/GUI/i7z_GUI.moc b/src/GUI/i7z_GUI.moc
deleted file mode 100644
index d2116af..0000000
--- a/src/GUI/i7z_GUI.moc
+++ /dev/null
@@ -1,93 +0,0 @@
-/****************************************************************************
-** Meta object code from reading C++ file 'i7z_GUI.cpp'
-**
-** Created: Fri Sep 13 10:43:03 2013
-**      by: The Qt Meta Object Compiler version 63 (Qt 4.8.4)
-**
-** WARNING! All changes made in this file will be lost!
-*****************************************************************************/
-
-#if !defined(Q_MOC_OUTPUT_REVISION)
-#error "The header file 'i7z_GUI.cpp' doesn't include <QObject>."
-#elif Q_MOC_OUTPUT_REVISION != 63
-#error "This file was generated using the moc from 4.8.4. It"
-#error "cannot be used with the include files from this version of Qt."
-#error "(The moc has changed too much.)"
-#endif
-
-QT_BEGIN_MOC_NAMESPACE
-static const uint qt_meta_data_MyWidget[] = {
-
- // content:
-       6,       // revision
-       0,       // classname
-       0,    0, // classinfo
-       1,   14, // methods
-       0,    0, // properties
-       0,    0, // enums/sets
-       0,    0, // constructors
-       0,       // flags
-       0,       // signalCount
-
- // slots: signature, parameters, type, tag, flags
-      10,    9,    9,    9, 0x08,
-
-       0        // eod
-};
-
-static const char qt_meta_stringdata_MyWidget[] = {
-    "MyWidget\0\0UpdateWidget()\0"
-};
-
-void MyWidget::qt_static_metacall(QObject *_o, QMetaObject::Call _c, int _id, void **_a)
-{
-    if (_c == QMetaObject::InvokeMetaMethod) {
-        Q_ASSERT(staticMetaObject.cast(_o));
-        MyWidget *_t = static_cast<MyWidget *>(_o);
-        switch (_id) {
-        case 0: _t->UpdateWidget(); break;
-        default: ;
-        }
-    }
-    Q_UNUSED(_a);
-}
-
-const QMetaObjectExtraData MyWidget::staticMetaObjectExtraData = {
-    0,  qt_static_metacall 
-};
-
-const QMetaObject MyWidget::staticMetaObject = {
-    { &QWidget::staticMetaObject, qt_meta_stringdata_MyWidget,
-      qt_meta_data_MyWidget, &staticMetaObjectExtraData }
-};
-
-#ifdef Q_NO_DATA_RELOCATION
-const QMetaObject &MyWidget::getStaticMetaObject() { return staticMetaObject; }
-#endif //Q_NO_DATA_RELOCATION
-
-const QMetaObject *MyWidget::metaObject() const
-{
-    return QObject::d_ptr->metaObject ? QObject::d_ptr->metaObject : &staticMetaObject;
-}
-
-void *MyWidget::qt_metacast(const char *_clname)
-{
-    if (!_clname) return 0;
-    if (!strcmp(_clname, qt_meta_stringdata_MyWidget))
-        return static_cast<void*>(const_cast< MyWidget*>(this));
-    return QWidget::qt_metacast(_clname);
-}
-
-int MyWidget::qt_metacall(QMetaObject::Call _c, int _id, void **_a)
-{
-    _id = QWidget::qt_metacall(_c, _id, _a);
-    if (_id < 0)
-        return _id;
-    if (_c == QMetaObject::InvokeMetaMethod) {
-        if (_id < 1)
-            qt_static_metacall(this, _c, _id, _a);
-        _id -= 1;
-    }
-    return _id;
-}
-QT_END_MOC_NAMESPACE

From c633ba108f691e8dd03b2c8b512d0cf5fda1fcc2 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Sun, 28 Apr 2019 16:30:57 -0400
Subject: [PATCH 2/2] Port to Qt5.

---
 src/GUI/i7z_GUI.pro | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/GUI/i7z_GUI.pro b/src/GUI/i7z_GUI.pro
index 4af29cc..c238364 100644
--- a/src/GUI/i7z_GUI.pro
+++ b/src/GUI/i7z_GUI.pro
@@ -3,12 +3,10 @@
 ######################################################################
 
 TEMPLATE = app
-TARGET = 
-DEPENDPATH += .
-INCLUDEPATH += .
+QT += widgets
 CONFIG += debug
 CFLAGS +=  #QMAKE_*FLAGS are where the gcc flags should be passed
-CXXFLAGS += 
+CXXFLAGS +=
 # Input
 SOURCES += i7z_GUI.cpp ../log.c
 LIBS += -lncurses
